src/tableau_server/tableau_server_objects.py

```diff
@@ -0,0 +1,372 @@
+import xmltodict
+from datetime import datetime
+from dataclasses import dataclass, asdict, astuple
+
+
+@dataclass
+class ServerObject:
```

Comment:
@JustinGrilli  what is defined as a server object? Is that any workbook or datasource?  A note or link to the docs on how that's defined would be great here.

Reply:
I'll add a note about it in the comment for this class.
But `ServerObject` can be any object we get when querying the Tableau API, i.e. Datasources, Workbooks, Users, Groups, Projects, etc.

This dataclass is just a class to be inherited by my other dataclasses, to provide some general functionality that my child classes are going to inherit (like those `__post__init` datatype conversions, and the `to_dict` method).